src/csharp/Grpc.IntegrationTesting/MetadataCredentialsTest.cs

```diff
@@ -103,6 +103,33 @@ public void MetadataCredentials_PerCall()
             client.UnaryCall(new SimpleRequest { }, new CallOptions(credentials: callCredentials));
         }
 
+        [Test]
+        public void MetadataCredentials_InterceptorLeavesMetadataEmpty()
+        {
+            var channelCredentials = ChannelCredentials.Create(TestCredentials.CreateSslCredentials(),
+                CallCredentials.FromInterceptor(new AsyncAuthInterceptor((context, metadata) => TaskUtils.CompletedTask)));
+            channel = new Channel(Host, server.Ports.Single().BoundPort, channelCredentials, options);
+            client = new TestService.TestServiceClient(channel);
+
+            var ex = Assert.Throws<RpcException>(() => client.UnaryCall(new SimpleRequest { }));
+            Assert.AreEqual(StatusCode.Unknown, ex.Status.StatusCode);
```

Comment:
not sure why this is expected to be `StatusCode.Unknown`. I would think `plugin_md_request_metadata_ready` should see there's no metadata, and run the callback with `GRPC_CREDENTIALS_OK`?